Sources/SwiftFormatWhitespaceLinter/WhitespaceLinter.swift

```diff
@@ -414,16 +427,124 @@ public class WhitespaceLinter {
       actions: actions
     )
   }
+
+  /// Returns the indentation that represents the indentation of the given whitespace, which is the
+  /// leading spacing for a line.
+  ///
+  /// A return value of nil indicates that there was no indentation.
+  private func indentation(of whitespace: String) -> WhitespaceIndentation {
+    if whitespace.count == 0 {
+      return .none
+    }
+
+    var orderedRuns: [(char: Character, count: Int)] = []
+    for char in whitespace {
+      // Any non-whitespace character indicates the end of the indentation whitespace.
+      guard char.isWhitespace else { break }
+
+      let lastRun = orderedRuns.last
+      if lastRun?.char == char {
+        orderedRuns[orderedRuns.endIndex - 1].count += 1
+      } else {
+        orderedRuns.append((char, 1))
+      }
+    }
+
+    let indents = orderedRuns.map { run in
+      // Assumes any non-tab whitespace character is some type of space.
+      return run.char == "\t" ? Indent.tabs(run.count) : Indent.spaces(run.count)
+    }
+    if indents.count == 1, let onlyIndent = indents.first {
+      return .homogeneous(onlyIndent)
+    }
+    return .heterogeneous(indents)
+  }
+}
+
+/// Describes the composition of the whitespace that creates an indentation for a line of code.
+enum WhitespaceIndentation: Equatable {
+  /// The line has no preceding whitespace, meaning there's no indentation.
+  case none
+
+  /// The line's leading whitespace consists of a single run of one kind of whitespace character.
+  case homogeneous(Indent)
+
+  /// The line's leading whitespace consists of multiple runs of different kinds of whitespace
+  /// characters.
+  case heterogeneous([Indent])
+}
+
+extension Indent {
+  /// Returns a string that describes the indentation in a human readable format, which is
+  /// appropriate for use in diagnostic messages.
+  var diagnosticDescription: String {
+    switch(self) {
```

Comment:
Remove parens around the `switch` argument here, and two (I think) places below please 🙂  (We should have a rule!)

Reply:
Done. The existing rule handled it; I just forgot to run the formatter. 😅